src/snowflake/snowpark/functions.py

```diff
@@ -3392,6 +3401,42 @@ def concat_ws(*cols: ColumnOrName, _emit_ast: bool = True) -> Column:
     return builtin("concat_ws", _emit_ast=_emit_ast)(*columns)
 
 
+@publicapi
+def concat_ws_ignore_nulls(
+    sep: str, *cols: ColumnOrName, _emit_ast: bool = True
+) -> Column:
+    """Concatenates two or more strings, or concatenates two or more binary values. Null values are ignored.
+
+    Args:
+        sep: The separator to use between the strings.
+
+    Examples::
+        >>> df = session.create_dataframe([
+        ...     ['Hello', 'World', None],
+        ...     [None, None, None],
+        ...     ['Hello', None, None],
+        ... ], schema=['a', 'b', 'c'])
+        >>> df.select(concat_ws_ignore_nulls(',', df.a, df.b, df.c)).show()
+        ----------------------------------------------------
+        |"CONCAT_WS_IGNORE_NULLS(',', ""A"",""B"",""C"")"  |
+        ----------------------------------------------------
+        |Hello,World                                       |
+        |                                                  |
+        |Hello                                             |
+        ----------------------------------------------------
+        <BLANKLINE>
+    """
+    columns = [_to_col_if_str(c, "concat_ws_ignore_nulls") for c in cols]
+    names = ",".join([c.get_name() for c in columns])
+    input_column_array = array_construct_compact(*columns, _emit_ast=_emit_ast)
+    reduced_result = builtin("reduce", _emit_ast=_emit_ast)(
+        input_column_array, lit(""), sql_expr(f"(l, r) -> l || '{sep}' || r")
+    )
+    return substring(reduced_result, 2, _emit_ast=_emit_ast).alias(
```

Comment:
you can use _emit_ast=False for both, we'll need to explicitly create the AST for this and then assign to the columnar expression returned.